src/crypto_tools/paillier/zk/composite_dlog.rs

```diff
@@ -321,7 +324,13 @@ mod tests {
 
         // Fail if a proof is very long
         bad_proof1.y = &proof1.y + dk.0.totient();
-        bad_proof2.y = &proof2.y + (dk.0.totient() << (CHALLENGE_K + SECURITY_PARAM_K_PRIME + 1));
+        // phi(N) is at least `MODULUS_MIN_SIZE` - 1
+        bad_proof2.y = &proof2.y
+            + (dk.0.totient()
+                << (CHALLENGE_K
+                    + SECURITY_PARAM_K_PRIME
+                    + (MODULUS_MAX_SIZE - MODULUS_MIN_SIZE)
+                    + 1));
```

Comment:
if this weren't a test I'd hound you about this expression. 😛

Reply:
Haha, I'll add more comments for when someone else has to touch this test.